src/graphene-box.c

```diff
@@ -436,18 +436,22 @@ graphene_box_is_empty (const graphene_box_t *box)
   return (isinff (vmin[0]) == 1 && isinff (vmin[1]) == 1 && isinff (vmin[2]) == 1) &&
          (isinff (vmax[0]) == -1 && isinff (vmax[1]) == -1 && isinff (vmax[2]) == -1);
 #else
-  graphene_simd4f_t neg_inf = graphene_simd4f_init (-INFINITY, -INFINITY, -INFINITY, 0.f);
-  graphene_simd4f_t pos_inf = graphene_simd4f_init (INFINITY, INFINITY, INFINITY, 0.f);
-
-  /* This is only every going to be valid for boxes that we have
-   * initialized ourselves, because we use the same values; the
-   * bitwise comparison will not hold for infinities generated by
-   * other operations
-   */
-  int min_cmp = memcmp (&box->min.value, &pos_inf, sizeof (graphene_simd4f_t));
-  int max_cmp = memcmp (&box->max.value, &neg_inf, sizeof (graphene_simd4f_t));
-
-  return min_cmp == 0 && max_cmp == 0;
+  float vmin[3], vmax[3];
+
+  graphene_simd4f_dup_3f (box->min.value, vmin);
```

Comment:
You could move these out of the `#ifdef` branch and save some duplicated lines (same below)